### PR TITLE
[IMP] point_of_sale,l10n_fr_pos_cert: relax french constraints and add order control feature

### PR DESCRIPTION
**Pull request by adgu-odoo**
Commit 1:

[IMP] point_of_sale: add order control feature
This commit adds the feature allowing the managers to control if changes
have been made to an order by a cashier. If changes have been made after
the order has been saved on the backend once, then the order is marked
as edited and a message is logged in the chatter of the order. At the
closing of the session, a message will be logged in the chatter of the
session informing which orders have been edited during the session.
An order is marked as "edited" if the quantity of a line is reduced
or a line is deleted.

Commit 2:

[IMP] l10n_fr_pos_cert: relax french constraints
In this commit, we relax teh french constrains on the PoS certification
module for France. This is due to a re-reading of the law and the
realization that we do not need to be so strict. The immutability and
security of the orders should be ensured after the validation of the
order, not before. We also allow the selection of a partner after the
validation of the order as it is not a immutable field regarding the
law.

task-id: 3874373

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by adgu-odoo** [comment]
Hello @odoo/rd-security can you check this ? The security warning is the use of Markup that I'm using to log message in the chatter. This would be great to have before the freeze. Thanks in advance

**Review by davidmonnom — APPROVED** [review]

**Comment by mart-e** [comment]
@adgu-odoo please insert only html in your Markup calls, not text

**Review by xmo-odoo — CHANGES_REQUESTED** [review]
> The security warning is the use of Markup that I'm using to log message in the chatter. This would be great to have before the freeze. Thanks in advance

All three notifications are for incorrect code. The first two warnings have no reason to use `Markup` at all, and the third one should only `Markup` the linebreak.

And all three should be passing the values they format in as arguments to the `_()` function not format post-facto.

**Comment by adgu-odoo** [comment]
Thank you for the comments

**Comment by davidmonnom** [comment]
@robodoo r+

**Comment by davidmonnom** [comment]
@robodoo rebase-ff

**Comment by davidmonnom** [comment]
@robodoo r+

**Comment by davidmonnom** [comment]
@robodoo r+

File: addons/l10n_fr_pos_cert/models/pos.py [hunk 1]
```diff
@@ -41,7 +38,8 @@ def open_frontend_cb(self):
         return super(pos_session, self).open_frontend_cb()
 
 
-ORDER_FIELDS = ['date_order', 'user_id', 'lines', 'payment_ids', 'pricelist_id', 'partner_id', 'session_id', 'pos_reference', 'sale_journal', 'fiscal_position_id']
+ORDER_FIELDS_BEFORE_17_4 = ['date_order', 'user_id', 'lines', 'payment_ids', 'pricelist_id', 'session_id', 'pos_reference', 'sale_journal', 'fiscal_position_id', 'partner_id']
+ORDER_FIELDS_FROM_17_4 = ['date_order', 'user_id', 'lines', 'payment_ids', 'pricelist_id', 'session_id', 'pos_reference', 'sale_journal', 'fiscal_position_id', 'pos_version']
```
Comment (legalsylvain) [inline addons/l10n_fr_pos_cert/models/pos.py:42]: With such change, a DB migrated from 17 to 18 will have hash control broken. and if we edit the document that certify that there is no change, the document will say that the orders are corrupted. 
is it deliberate ?

**Pull request closed without merging**